packages/signals_flutter/lib/src/signals/core/signal.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:flutter/material.dart';
+
+import '../../../signals_flutter.dart';
+
+final _signals = <WeakReference<Element>, Map<int, Signal>>{};
+
+/// Create and watch a signal in a build method
+///
+/// ```dart
+/// class MyWidget extends ... {
+///  @override
+///  Widget build(BuildContext context) {
+///    final count = useSignal(context, 0);
+///    return Row(
+///     children: [
+///       IconButton(icon: Icon(Icons.remove), onPressed: () => count.value--),
+///       Text(count.value.toString()),
+///       IconButton(icon: Icon(Icons.add), onPressed: () => count.value++),
+///    ],
+///   );
+///  }
+/// }
+/// ```
+/// 
+/// If you are defining multiple signals in a widget with the same type you
+/// need to give it a label to differentiate them:
+/// 
+/// ```dart
+/// class MyWidget extends ... {
+///  @override
+///  Widget build(BuildContext context) {
+///    final count = useSignal(context, 0);
+///    final count2 = useSignal(context, 0, debugLabel: 'count2');
+///    return Row(
+///     children: [
+///       IconButton(icon: Icon(Icons.remove), onPressed: () => count.value--),
+///       Text(count.value.toString()),
+///       IconButton(icon: Icon(Icons.add), onPressed: () => count.value++),
+///    ],
+///   );
+///  }
+/// }
+/// ```
+Signal<T> useSignal<T>(
+  BuildContext context,
+  T value, {
+  String? debugLabel,
+  SignalEquality<T>? equality,
+}) {
+  assert(context is Element);
+  final key = (value, debugLabel).hashCode;
```

Comment:
Wouldn't it be safer to include `T` in the key? This way, `useSignal<String?>(null)` and `useSignal<int?>(null)` won't collide, for example.

Reply:
Thats a great callout. Going to pull out these to another draft PR